sui/deploy-utils.js

```diff
@@ -59,130 +70,49 @@ async function upgradePackage(client, keypair, packageName, packageConfig, build
-    await builder.publishPackageAndTransferCap(packageName, address);
-    const tx = builder.tx;
-    tx.setSender(address);
-    const txBytes = await tx.build({ client });
-
-    if (offline) {
-        options.txBytes = txBytes;
-    } else {
-        if (prompt(`Proceed with deployment on ${chain.name}?`, options.yes)) {
-            return;
-        }
-
-        const signature = (await keypair.signTransaction(txBytes)).signature;
-        const publishTxn = await client.executeTransactionBlock({
-            transactionBlock: txBytes,
-            signature,
-            options: {
-                showEffects: true,
-                showObjectChanges: true,
-                showEvents: true,
-            },
-        });
-
-        packageConfig.address = (publishTxn.objectChanges?.find((a) => a.type === 'published') ?? []).packageId;
-        const objectChanges = publishTxn.objectChanges.filter((object) => object.type === 'created');
-        packageConfig.objects = {};
-
-        for (const object of objectChanges) {
-            const array = object.objectType.split('::');
-            const objectName = array[array.length - 1];
+        contractConfig.address = packageId;
+        const [upgradeCap] = getObjectIdsByObjectTypes(result, [`${suiPackageAddress}::package::UpgradeCap`]);
+        contractConfig.objects.UpgradeCap = upgradeCap;
 
-            if (objectName) {
-                packageConfig.objects[objectName] = object.objectId;
-            }
-        }
-
-        printInfo(`${packageName} deployed`, JSON.stringify(packageConfig, null, 2));
+        printInfo('Transaction Digest', JSON.stringify(result.digest, null, 2));
+        printInfo(`${packageName} Upgraded Address`, packageId);
     }
 }
 
-async function processCommand(chain, options) {
-    const [keypair, client] = getWallet(chain, options);
-    const { upgrade, packageName, packageDependencies, offline, txFilePath } = options;
-
-    printInfo('Wallet address', keypair.toSuiAddress());
-
-    if (!chain.contracts[packageName]) {
-        chain.contracts[packageName] = {};
-    }
-
-    const contractsConfig = chain.contracts;
-    const packageConfig = contractsConfig?.[packageName];
-
-    validateParameters({ isNonEmptyString: { packageName } });
-
-    if (packageDependencies) {
-        for (const dependencies of packageDependencies) {
-            const packageId = contractsConfig[dependencies]?.address;
-            updateMoveToml(dependencies, packageId);
-        }
-    }
-
-    const builder = new TxBuilder(client);
-
-    if (upgrade) {
-        await upgradePackage(client, keypair, packageName, packageConfig, builder, options);
-    } else {
-        await deployPackage(chain, client, keypair, packageName, packageConfig, builder, options);
-    }
-
-    if (offline) {
-        validateParameters({ isNonEmptyString: { txFilePath } });
-
-        const txB64Bytes = toB64(options.txBytes);
+const getDeployGatewayOptions = () => {
```

Comment:
this should be moved to deploy-contract where gateway deployment is defined instead of utils

Reply:
hmm yeah, but need to move the `addDeployOptions` function (it was defined below this function) to deploy-contract as well because then i have to import `getDeployGatewayOptions` from `deploy-contract.js` in the utils file which is weird.